### PR TITLE
feat(ui): TE-2363 allow user to opt in to be notified for historical anomalies when creating a subscription

### PR DESCRIPTION
**Pull request by nalin-patidar**
…cription

#### Issue(s)
https://startree.atlassian.net/browse/TE-2363

#### Description

Added the ability for a user to subscribe to notification for historical anomalies.

#### Screenshots

https://github.com/startreedata/thirdeye/assets/174305
Screenshot 2024-07-05 at 7 59 35 PM
624/132a709a-b67b-4577-bdbf-6929598dfda0

### CODE REVIEW

**Review by alcatraz627 — APPROVED** [review]

**Comment by alcatraz627** [comment]
@nalin-patidar PR looks good, but the title check is failing. The capital "A" in the title needs to be changed to lowercase + since only one commit in this PR, will need to `git commit --amend` and `git push -f` to change the same in the commit message as well.

Read about it at https://github.com/startreedata/thirdeye/blob/master/thirdeye-ui/CONTRIBUTING.md

**Comment by cyrilou242** [comment]
@alcatraz627 
> PR looks good, but the title check is failing. The capital "A" in the title needs to be changed to lowercase +

could we remove such constraint? I don't see the value in lower/upper case checks

**Review by cyrilou242 — CHANGES_REQUESTED** [review]
let's add a tooltip that explaines the feature.
I provided the tooltip message in the ticket.

You should be able to find some tooltip code in the codebase.

Screenshot 2024-07-05 at 12 00 38

**Comment by alcatraz627** [comment]
> could we remove such constraint? I don't see the value in lower/upper case checks

@cyrilou242 the thirdeye-ui conventions are based on whatever was set up for `startree-ui`. Though we are trying to maintain consistency for all UI projects, we might be able to get rid of such minor details like capitalization.

File: thirdeye-ui/src/app/components/subscription-group-wizard/subscription-group-details/subscription-group-details.component.tsx [hunk 1]
```diff
@@ -23,6 +23,8 @@ export const SubscriptionGroupDetails: FunctionComponent<SubscriptionGroupDetail
         const propertiesFormValues = {
             cron: subscriptionGroup.cron,
             name: subscriptionGroup.name,
+            notifyHistoricalAnomalies:
+                !!subscriptionGroup.notifyHistoricalAnomalies,
```
Comment (jayeshchoudhary) [inline thirdeye-ui/src/app/components/subscription-group-wizard/subscription-group-details/subscription-group-details.component.tsx:27]: nit: for better readability 
```suggestion
            notifyHistoricalAnomalies:
                Boolean(subscriptionGroup.notifyHistoricalAnomalies),
```

Comment (nalin-patidar) [reply thirdeye-ui/src/app/components/subscription-group-wizard/subscription-group-details/subscription-group-details.component.tsx]: Done

File: thirdeye-ui/src/app/components/info-label/info-label.component.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+*/
+import React, { FunctionComponent } from "react";
+import { InfoLabelProps } from "./info-label.interfaces";
+import { Grid, Tooltip, Typography } from "@material-ui/core";
+import InfoIconOutlined from "@material-ui/icons/InfoOutlined";
+
+export const InfoLabel: FunctionComponent<InfoLabelProps> = ({
+    label,
+    tooltipText = "",
+}) => {
```
Comment (jayeshchoudhary) [inline thirdeye-ui/src/app/components/info-label/info-label.component.tsx:30]: we should have similar component already?
how are we showing tooltips now?

**Comment by jayeshchoudhary** [comment]
I also see the label and info icon are not centred vertically
is the pr screenshot update?

File: thirdeye-ui/src/app/components/info-label/info-label.component.tsx (hunk 2 above)
Comment (nalin-patidar) [reply thirdeye-ui/src/app/components/info-label/info-label.component.tsx:30]: I have commented about this in top of this file

**Pull request merged**